x/payment/keeper/storage_fee_charge.go

```diff
@@ -52,50 +52,94 @@ func (k Keeper) ApplyStreamRecordChanges(ctx sdk.Context, streamRecordChanges []
 func (k Keeper) ApplyUserFlowsList(ctx sdk.Context, userFlowsList []types.UserFlows) (err error) {
 	userFlowsList = k.MergeUserFlows(userFlowsList)
 	currentTime := ctx.BlockTime().Unix()
-	var streamRecordChanges []types.StreamRecordChange
+
 	for _, userFlows := range userFlowsList {
 		from := userFlows.From
 		streamRecord, found := k.GetStreamRecord(ctx, from)
 		if !found {
 			streamRecord = types.NewStreamRecord(from, currentTime)
 		}
-		// calculate rate changes in flowChanges
-		totalRate := sdk.ZeroInt()
-		for _, flowChange := range userFlows.Flows {
-			streamRecordChanges = append(streamRecordChanges, *types.NewDefaultStreamRecordChangeWithAddr(sdk.MustAccAddressFromHex(flowChange.ToAddress)).WithRateChange(flowChange.Rate))
-			totalRate = totalRate.Add(flowChange.Rate)
-		}
-		streamRecordChange := types.NewDefaultStreamRecordChangeWithAddr(from).WithRateChange(totalRate.Neg())
-		// storage fee preview
-		if ctx.IsCheckTx() {
-			reserveTime := k.GetParams(ctx).VersionedParams.ReserveTime
-			changeRate := totalRate.Neg()
-			event := &types.EventFeePreview{
-				Account: from.String(),
-				Amount:  changeRate.Mul(sdkmath.NewIntFromUint64(reserveTime)).Abs(),
+		if streamRecord.Status == types.STREAM_ACCOUNT_STATUS_ACTIVE {
```

Comment:
this part is a bit long, can we abstract these codes to a separate function.

Reply:
yes, created separated functions